catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/TransformersCommand.java

```diff
@@ -52,95 +59,196 @@
+  )
+  private boolean inputOption = false;
+
+  @Option(
+    name = "--metacard",
+    required = false,
+    aliases = {"-m", "-metacard"},
+    multiValued = false,
+    description = "Displays only metacard transformers"
+  )
+  private boolean metacardOption = false;
+
+  @Option(
+    name = "--all",
+    required = false,
+    aliases = {"-a", "-all"},
+    multiValued = false,
+    description = "Displays all the properties of the desired transformers"
+  )
+  private boolean allOption = false;
+
   @Override
   protected Object executeWithSubject() throws Exception {
+    if (!inputOption && !metacardOption) {
+      inputOption = true;
+      metacardOption = true;
+    }
+
+    if (inputOption) {
+      printTransformers(InputTransformer.class, INPUT);
+    }
+    if (metacardOption) {
+      printTransformers(MetacardTransformer.class, METACARD);
+    }
 
-    // ServiceReferences are needed to get property information from InputTransformers
-    List<ServiceReference<InputTransformer>> serviceReferences =
-        (List<ServiceReference<InputTransformer>>)
-            bundleContext.getServiceReferences(InputTransformer.class, "(id=*)");
+    return null;
+  }
+
+  private void printTransformers(Class transformerClass, String type) throws Exception {
+
+    Collection<ServiceReference> sref =
+        bundleContext.getServiceReferences(transformerClass, FILTER);
 
     List<TransformerProperties> transformersProperties =
-        serviceReferences.stream().map(TransformerProperties::new).collect(Collectors.toList());
+        sref.stream().map(TransformerProperties::new).collect(Collectors.toList());
 
     int activeTransformers = transformersProperties.size();
 
     if (activeTransformers == 0) {
-      console.printf("%s%n%n", NO_ACTIVE_TRANSFORMERS);
-      return null;
+      console.printf(getNoActiveString(type));
+      return;
     }
 
-    console.printf("%n%s%d%n%s%n%n", ACTIVE_TRANSFORMERS_HEADER, activeTransformers, LINE);
+    console.printf(getActiveNumString(type, activeTransformers));
 
     Iterator<TransformerProperties> tpIterator = transformersProperties.iterator();
     TransformerProperties tp;
 
     while (tpIterator.hasNext()) {
       tp = tpIterator.next();
-      console.printf("%s", tp.printProperties());
+
+      if (allOption) {
+        console.printf("%s", tp.printAllProperties());
+      } else {
+        console.printf("%s", tp.printDefaultProperties());
+      }
 
       if (tpIterator.hasNext()) {
         console.printf("%n%s%n", StringUtils.repeat(LINE, 3));
       }
       console.printf("%n");
     }
-
-    return null;
   }
 
-  private static class TransformerProperties {
-
-    private String id;
+  protected static String getNoActiveString(String type) {
+    return String.format("%s%n%n", "There are no active " + type + " transformers");
+  }
 
-    private String schema;
+  protected static String getActiveNumString(String type, int activeTransformers) {
```

Comment:
Nit: looks like some more of these fields used in `TransformersCommandTest` need `@VisibleForTesting`:
```java
import static org.codice.ddf.commands.catalog.TransformersCommand.FILTER;
import static org.codice.ddf.commands.catalog.TransformersCommand.INPUT;
import static org.codice.ddf.commands.catalog.TransformersCommand.METACARD;
import static org.codice.ddf.commands.catalog.TransformersCommand.getActiveNumString;
import static org.codice.ddf.commands.catalog.TransformersCommand.getNoActiveString;
```
I might be wrong, but I though instead of making fields/methods visible and importing them from the class that's being tested, it's better to redefine the fields/methods in the unit test class (e.g. `assertThat(consoleOutput.getOutput(), containsString("Active Input transformers: ..."));` instead of `assertThat(consoleOutput.getOutput(), containsString(getActiveNumString(INPUT, notNullIds)));`).